libcalico-go/lib/backend/k8s/resources/resources.go

```diff
@@ -175,6 +176,17 @@ func ConvertCalicoResourceToK8sResource(resIn Resource) (Resource, error) {
 	// Name, Namespace, ResourceVersion, UID, and Annotations (built above).
 	meta := &metav1.ObjectMeta{}
 	meta.Name = rom.GetName()
+	switch resIn.GetObjectKind().GroupVersionKind().Kind {
+	// For NetworkPolicy and GlobalNetworkPolicy, we need to prefix the name with the tier name.
```

Comment:
```suggestion
	// For NetworkPolicy and GlobalNetworkPolicy, we need to prefix the name with the tier name.
	// This ensures two policies with the same name, but in different tiers, do not resolve to the same backing object.
```

Reply:
A bit of a nit, but could you move the switch statement to the end of the metadata populating section? It's breaking up a bunch of similar lines and makes the code a tiny bit harder to follow. 

e.g.,

```go
meta.Name = rom.GetName()
meta.Namespace = rom.GetNamespace()
meta.ResourceVersion = rom.GetResourceVersion()

switch resIn.GetObjectKind().GroupversionKind().Kind() {
 . . .
```